--- a/ARM/ARMinstructions.sinc
+++ b/ARM/ARMinstructions.sinc
@@ -182,6 +182,9 @@
     c1621=(16,21)
     c1620=(16,20)
     c1619=(16,19)
+    bankRegM0=(16,19)
+    bankRegM1=(16,19)
+    mrsBankIdx=(16,19)
     c1618=(16,18)
     c1617=(16,17)
     c1616=(16,16)
@@ -700,6 +703,9 @@
 attach names [ ibOption ] [ opt0 opt1 opt2 opt3 opt4 opt5 opt6 opt7 opt8 opt9 opt10 opt11 opt12 opt13 opt14 SY ];
 attach names [ dbOption ] [ opt0 opt1 OSHST OSH opt4 opt5 NSHST NSH opt8 opt9 ISHST ISH opt12 opt13 ST SY ];
 
+attach names [ bankRegM0 ] [ r8_usr r9_usr r10_usr r11_usr r12_usr sp_usr lr_usr _ r8_fiq r9_fiq r10_fiq r11_fiq r12_fiq sp_fiq lr_fiq _ ];
+attach names [ bankRegM1 ] [ lr_irq sp_irq lr_svc sp_svc lr_abt sp_abt lr_und sp_und _ _ _ _ lr_mon sp_mon elr_hyp sp_hyp ];
+
 macro addflags(op1,op2) {
  tmpCY = carry(op1,op2);
  tmpOV = scarry(op1,op2);
@@ -797,6 +803,7 @@
 }
 
 ####################
+define pcodeop readBankedReg;
 define pcodeop coproc_moveto_Main_ID;
 define pcodeop coproc_moveto_Cache_Type;
 define pcodeop coproc_moveto_TCM_Status;
@@ -4261,6 +4268,19 @@
   Rd = spsr;
 }
 
+:mrs^COND Rd,bankRegM0 	is $(AMODE) & ARMcond=1 & COND & c2327=2 & c2222=0 & c2021=0 & bankRegM0 & mrsBankIdx & Rd & c1011=0 & c0909=1 & c0808=0 & c0407=0 & c0003=0
+{
+  build COND;
+  Rd = readBankedReg(mrsBankIdx:1);
+}
+
+:mrs^COND Rd,bankRegM1 	is $(AMODE) & ARMcond=1 & COND & c2327=2 & c2222=0 & c2021=0 & bankRegM1 & mrsBankIdx & Rd & c1011=0 & c0909=1 & c0808=1 & c0407=0 & c0003=0
+{
+  build COND;
+  local sysm:1 = mrsBankIdx + 16;
+  Rd = readBankedReg(sysm);
+}
+
 :msr^COND cpsrmask,shift1 	is $(AMODE) & ARMcond=1 & COND & c2027=50 & cpsrmask & c1215=15 & c2627=0 & shift1
 {
   build COND;
